cf/test/test_Field.py

```diff
@@ -743,36 +743,32 @@ def test_Field_cumsum(self):
         self.assertIsNone(g.cumsum(2, inplace=True))
         self.assertTrue(g.equals(h, verbose=2))
 
+        # Check that a new cell method that has been added
+        cell_methods = h.cell_methods(todict=True)
+        self.assertEqual(len(cell_methods), len(f.cell_methods()) + 1)
+        _, cm = cell_methods.popitem()
+        self.assertEqual(cm.method, "sum")
+        self.assertEqual(cm.axes, (h.get_data_axes()[2],))
+
+        # Check the dimension coordinate bounds
+        fx = f.dimension_coordinate("X")
+        hx = h.dimension_coordinate("X")
+        self.assertTrue((hx.lower_bounds == fx.lower_bounds[0]).all())
+        self.assertTrue((hx.upper_bounds == fx.upper_bounds).all())
+
+        a = f.array
         for axis in range(f.ndim):
-            a = numpy.cumsum(f.array, axis=axis)
-            self.assertTrue((f.cumsum(axis=axis).array == a).all())
+            b = np.cumsum(a, axis=axis)
+            self.assertTrue((f.cumsum(axis=axis).array == b).all())
 
         f[0, 0, 3] = cf.masked
         f[0, 2, 7] = cf.masked
 
+        a = f.array
         for axis in range(f.ndim):
-            a = f.array
-            a = numpy.cumsum(a, axis=axis)
+            b = np.cumsum(a, axis=axis)
             g = f.cumsum(axis=axis)
-            self.assertTrue(cf.functions._numpy_allclose(g.array, a))
-
-        for axis in range(f.ndim):
-            g = f.cumsum(axis=axis, masked_as_zero=True)
-
-            a = f.array
-            mask = a.mask
-            a = a.filled(0)
-            a = numpy.cumsum(a, axis=axis)
-            size = a.shape[axis]
-            shape = [1] * a.ndim
-            shape[axis] = size
-            new_mask = numpy.cumsum(mask, axis=axis) == numpy.arange(
-                1, size + 1
-            ).reshape(shape)
-            a = numpy.ma.array(a, mask=new_mask, copy=False)
-            self.assertTrue(
-                cf.functions._numpy_allclose(g.array, a, verbose=2)
-            )
+            self.assertTrue(cf.functions._numpy_allclose(g.array, b))
```

Comment:
Although I see we use this form of what is, I assume, a corresponding assertion in numerous other places in this module and `test_Data`, in other places we use `numpy.testing.assert_allclose` and I think that might be a more transparent form to use for testing, so how about, instead:

```suggestion
            numpy.testing.assert_allclose(g.array, b)
```

and make the equivalent conversion for other usage of `self.assertTrue(cf.functions._numpy_allclose(<stuff>))`? Or could go vice versa, but would be nice to use one or the other consistently, I think. (If we want to do this ultimately, it can all be done in another dedicated PR, though, not here.)

Reply:
I think that this is one for another day, as `numpy.testing.assert_allclose` has different functionality to `cf.functions._numpy_allclose`

Reply:
Fair enough, I was assuming they were similar but should really have checked before commenting :slightly_smiling_face: All good, marking as resolved!